src/wrapper/jnienv.rs

```diff
@@ -1033,7 +1033,7 @@ impl<'local> JNIEnv<'local> {
     /// [`GlobalRef`] / [`Self::make_global`].
     pub fn with_local_frame<F, T, E>(&mut self, capacity: i32, f: F) -> std::result::Result<T, E>
     where
-        F: FnOnce(&mut JNIEnv) -> std::result::Result<T, E>,
+        F: FnOnce(&mut JNIEnv<'local>) -> std::result::Result<T, E>,
```

Comment:
It's very important that the `JNIEnv` that is passed to `f` must not have the same lifetime as the outer `JNIEnv` so I don't think this change is safe.

The purpose of the `JNIEnv` lifetime is to represent the JNI local stack frame that local references belong to and since this API creates a new JNI stack frame (via `PushLocalFrame`) then the `JNIEnv` needs to have a new, unique lifetime, otherwise it would be possible to move a local reference (such as `JObject<'local>` or `JString<'local>` out of the given closure back to the caller, where they would effectively become invalid pointers.

Reply:
E.g. I guess this change would let you do something like:

```rust
let mut outer_ref: Option<JString<'outer>> = None;
env.with_local_frame(5, |env| {
    let local_ref = env.new_string("foo").unwrap();
    outer_ref.replace(local_ref);
    Ok(())
}).unwrap();
```

and then when `with_local_frame` returns if you would try and access `outer_ref` it would contain an invalid pointer for a local reference that was freed when `PopLocalFrame` was called within `with_local_frame()`.